File: sdk/python/aragora/namespaces/calibration.py
"""
Calibration Namespace API

Provides access to agent calibration and leaderboard data.
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from ..client import AragoraAsyncClient, AragoraClient


class CalibrationAPI:
    """Synchronous Calibration API for agent calibration data."""

    def __init__(self, client: AragoraClient):
        self._client = client

    def get_leaderboard(self) -> dict[str, Any]:
        """Get calibration leaderboard.

        Returns:
            Leaderboard with agent calibration scores and rankings.
        """
        return self._client.request("GET", "/api/v1/calibration/leaderboard")

    def get_visualization(self) -> dict[str, Any]:
        """Get calibration visualization data.

        Returns:
            Visualization data for calibration metrics.
        """
        return self._client.request("GET", "/api/v1/calibration/visualization")


class AsyncCalibrationAPI:
    """Asynchronous Calibration API for agent calibration data."""

    def __init__(self, client: AragoraAsyncClient):
        self._client = client

    async def get_leaderboard(self) -> dict[str, Any]:
        """Get calibration leaderboard.

        Returns:
            Leaderboard with agent calibration scores and rankings.
        """
        return await self._client.request("GET", "/api/v1/calibration/leaderboard")

    async def get_visualization(self) -> dict[str, Any]:
        """Get calibration visualization data.

        Returns:
            Visualization data for calibration metrics.
        """
        return await self._client.request("GET", "/api/v1/calibration/visualization")

File: sdk/typescript/src/namespaces/calibration.ts
/**
 * Calibration Namespace API
 *
 * Provides access to agent calibration leaderboards, curves, and summaries.
 */

export interface CalibrationLeaderboardEntry {
  agent: string;
  score: number;
  samples?: number;
  rank?: number;
}

export interface CalibrationVisualization {
  agents: Array<{
    name: string;
    calibration_curve: Array<{ confidence: number; accuracy: number; count: number }>;
    overall_score: number;
  }>;
}

export interface CalibrationCurve {
  agent: string;
  data_points: Array<{
    confidence: number;
    accuracy: number;
    sample_count: number;
  }>;
  overall_score: number;
  total_predictions: number;
}

export interface CalibrationSummary {
  agent: string;
  calibration_score: number;
  overconfidence_bias: number;
  underconfidence_bias: number;
  total_predictions: number;
  accuracy: number;
  last_updated?: string;
}

export interface AgentCalibration {
  agent: string;
  scores: Record<string, number>;
  history?: Array<{ timestamp: string; score: number }>;
}

interface CalibrationClientInterface {
  request<T = unknown>(
    method: string,
    path: string,
    options?: { params?: Record<string, unknown> }
  ): Promise<T>;
}

export class CalibrationAPI {
  constructor(private client: CalibrationClientInterface) {}

  /**
   * Get the calibration leaderboard.
   */
  async getLeaderboard(params?: {
    limit?: number;
  }): Promise<{ leaderboard: CalibrationLeaderboardEntry[]; total?: number }> {
    return this.client.request('GET', '/api/v1/calibration/leaderboard', {
      params: params as Record<string, unknown>,
    });
  }

  /**
   * Get calibration visualization data for all agents.
   */
  async getVisualization(): Promise<CalibrationVisualization> {
    return this.client.request('GET', '/api/v1/calibration/visualization');
  }

  /**
   * Get the calibration curve for a specific agent.
   */
  async getCurve(agentName: string): Promise<CalibrationCurve> {
    return this.client.request('GET', `/api/v1/agent/${agentName}/calibration-curve`);
  }

  /**
   * Get the calibration summary for a specific agent.
   */
  async getSummary(agentName: string): Promise<CalibrationSummary> {
    return this.client.request('GET', `/api/v1/agent/${agentName}/calibration-summary`);
  }

  /**
   * Get general calibration scores for a specific agent.
   */
  async getAgentCalibration(agentName: string): Promise<AgentCalibration> {
    return this.client.request('GET', `/api/v1/agent/${agentName}/calibration`);
  }
}
